### DISCUSSION TITLE
Episode 169: thrust::unique_count and The Algorithm Advisor

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#61] [Episode]
Episode Link: https://adspthepodcast.com/2024/02/16/Episode-169.html

Feel free to write comments or ask questions 🙂

**Comment by flandweber** [comment]
I suspect that in Haskell many functions with the same semantics will compile to similar programs.
Nevertheless, I think there could be great value in an algorithm advisor for Haskell to have programmers simplify their code.
Does something like this exist?
I couldn't find anything, but I'm also not sure what to search for.

**Reply by codereport** [reply]
I know of no "algorithm advisor" in Haskell but the closest thing I can think of is Hoogle+: 

* https://www.youtube.com/watch?v=rjz_m8aTul8
* https://github.com/TyGuS/hoogle_plus
